### PR TITLE
Vt 7

### PR DESCRIPTION
**Pull request by LINDVIOR18**
Enable JWT (Json Web Token), Filter, PreAuthorization

### CODE REVIEW

File: src/main/java/com/vesta/config/security/JWTFilter.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Autowired
+    private TokenService tokenService;
+
+    @Autowired
+    private UserService userService;
+
+    public JWTFilter(TokenService tokenService) {
+
+        this.tokenService = tokenService;
```
Comment (ccaraian) [inline src/main/java/com/vesta/config/security/JWTFilter.java]: Why you do this `this.tokenService = tokenService` ?  when you already have 
```
@Autowired
private TokenService tokenService;
```
Use only one, @Autowired or this.tokenService = tokenService;

File: src/main/java/com/vesta/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    @PostMapping("/login")
+    public Token login(@RequestBody AccountCredential accountCredential) {
+        return userService.login(accountCredential);
+    }
+
+    @GetMapping("/{id}")
+    @PreAuthorize("hasRole('ROLE_user')")
+    public UserView getById(@PathVariable Long id){
+        SecurityContextHolder.getContext().getAuthentication();
```
Comment (ccaraian) [inline src/main/java/com/vesta/controller/UserController.java:29]: remove this line, it was added only for test

File: src/main/java/com/vesta/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    @Autowired
+    private UserViewConverter userViewConverter;
+
+    @PostMapping("/login")
+    public Token login(@RequestBody AccountCredential accountCredential) {
+        return userService.login(accountCredential);
+    }
+
+    @GetMapping("/{id}")
+    @PreAuthorize("hasRole('ROLE_user')")
```
Comment (ccaraian) [inline src/main/java/com/vesta/controller/UserController.java]: remove this line, it was added only for test

File: src/main/java/com/vesta/config/security/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Autowired
+    private JWTFilter jwtFilter;
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .antMatchers(HttpMethod.POST, "/user/login").permitAll()
+                .antMatchers("/v2/api-docs").permitAll()
```
Comment (vadimeladii) [inline src/main/java/com/vesta/config/security/SecurityConfig.java]: Use this:
private static final String[] AUTH_WHITELIST = {
            // -- swagger ui
            "/swagger-resources/**",
            "/swagger-ui.html",
            "/v2/api-docs",
            "/webjars/**",
            "/api/swagger.json"
    };

File: src/main/java/com/vesta/VestaApiApplication.java [hunk 5]
```diff
@@ -2,8 +2,9 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration;
 
-@SpringBootApplication
+@SpringBootApplication(exclude = { SecurityAutoConfiguration.class })
```
Comment (vadimeladii) [inline src/main/java/com/vesta/VestaApiApplication.java]: It's not necessary. Remove SecurityAutoConfiguration

**Review by vadimeladii — APPROVED** [review]

**Pull request merged**